File: doc/src/codes/Users_Manual/advanced_examples/custom_modulation_0.cs
public class Burst : Modulation
{
    private readonly int _length;

    public Burst(int length) : base(SamplingConfiguration.FromFrequency(4e3))
    {
        _length = length;
    }

    public override EmitIntensity[] Calc()
    {
        var buf = Enumerable.Repeat<EmitIntensity>(EmitIntensity.Min, _length).ToArray();
        buf[0] = EmitIntensity.Max;
        return buf;
    }
}

File: doc/src/codes/Users_Manual/advanced_examples/custom_modulation_0.rs
# extern crate autd3;
# extern crate tokio;
# extern crate autd3_driver;
use autd3::{
    derive::Modulation,
};
use autd3_driver::derive::prelude::*;

#[derive(Modulation)]
pub struct Burst {
    config: SamplingConfiguration,
}

impl Burst {
    pub fn new() -> Self {
        Self { config: SamplingConfiguration::from_frequency(4e3).unwrap() }
    }
}

impl Modulation for Burst {
    fn calc(&self) -> Result<Vec<EmitIntensity>, AUTDInternalError> {
        Ok((0..4000)
            .map(|i| if i == 3999 { EmitIntensity::MAX } else { EmitIntensity::MIN })
            .collect())
    }
}
# fn main() { 
# }